android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMKeyboardDownloaderActivity.java

```diff
@@ -344,8 +342,13 @@ protected int downloadNonKMPKeyboard(String remoteUrl) throws Exception {
 
         result = FileUtils.download(context, url, destination, filename);
         if (result < 0) {
-          ret = -1;
-          break;
+          if (FileUtils.hasFontExtension(url)) {
+            // Propogate warning about font failing to download
+            ret = 2;
```

Comment:
Should probably add this in the function's documentation comment as a possibility.

Also, maybe make it a custom exception instead?  Then catch that exception later where we expect to see it as a possibility.  That's probably better design than the old C-era error code strat.

Reply:
I've updated the header comment. Hesitant to make it a custom exception though.
a) It's a "warning" condition where the keyboard downloads but not the font
b) The async task is meant to use `res` for the "result".

Reply:
Ah, if it's part of something async, that makes sense then.